ml-agents/mlagents/trainers/torch/layers.py

```diff
@@ -206,7 +207,19 @@ def forward(
         # We don't use torch.split here since it is not supported by Barracuda
         h0 = memories[:, :, : self.hidden_size].contiguous()
         c0 = memories[:, :, self.hidden_size :].contiguous()
+
+        if exporting_to_onnx.is_exporting():
+            # This transpose is needed both at input and output of the LSTM when
+            # exporting because ONNX will expect (sequence_len, batch, memory_size)
+            # instead of (batch, sequence_len, memory_size)
+            h0 = torch.transpose(h0, 0, 1)
```

Comment:
I actually think we should transpose it before the split into `(h0, c0)` as it will be marginally faster and will be symmetrical with the transpose on the output below. But it's not a dealbreaker.